plugin/nginx_plus_collectd.py

```diff
@@ -772,6 +801,25 @@ def get_processes(self):
         '''
         return self._send_get(self.processes_url)
 
+    def detect_api_type(self):
+        '''
+        Detects the API type of the Nginx-plus(Newer or Legacy) and initialize the endpoints accordingly
+        '''
+        detected_api_type = self._get_api_type()
```

Comment:
I think this should only be done if the user hasn't specified anything.

Reply:
The purpose of this is the same as [this](https://github.com/signalfx/collectd-nginx-plus/pull/8#discussion_r749912399).

Reply:
Hello @rmfitzpatrick,

I completely agree with your above-mentioned point but, as per the latest design(explained [here](https://github.com/signalfx/collectd-nginx-plus/pull/8#discussion_r757691244)), this entire function has been removed. Hence, no changes are required.

Reply:
This is still an issue since you are potentially still disregarding user input by blindly "detecting"

Reply:
In the last pushed commit, we have made the change. Now, we are only detecting the NGINX+ API type (versioned or legacy) if `APIVersion` is not provided by the user.